hpOneView/connection.py

```diff
@@ -479,8 +479,11 @@ def change_initial_password(self, newPassword):
     # Login/Logout to/from appliance
     ###########################################################################
     def login(self, cred, verbose=False):
-        if self._validateVersion is False:
-            self.validateVersion()
+        try:
+            if self._validateVersion is False:
+                self.validateVersion()
+        except Exception as e:
+            raise HPOneViewException(str(e))
```

Comment:
Why do we need to cover every exception?

An InterruptedException, for instance, should not be re-raised as a HPOneViewException, these are not SDK errors

Reply:
Sure, what about the `no route to host` or `timeout` exceptions? Should we specifically seek those out?

Reply:
Maybe you should try:
```python
raise HPOneViewException(str(e)) from e
```
But I don't know if future will handle it well.

Reply:
I've just tried the `raise_from` from future, but it does not appear to handle it as it should:

On python2:
```
vagrant@ecosystem-centos7:~/dev/python-hpOneView/examples  (enhancement/catch-http-lib-exceptions-as-hponeviewexceptions)
$ python fc_networks.py
Traceback (most recent call last):
  File "fc_networks.py", line 54, in <module>
    oneview_client = OneViewClient(config)
  File "/home/vagrant/dev/python-hpOneView/hpOneView/oneview_client.py", line 119, in __init__
    self.__connection.login(config["credentials"])
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 491, in login
    raise_from(HPOneViewException('Failure during login attempt.'), e)
  File "/usr/local/lib/python2.7/site-packages/future/utils/__init__.py", line 454, in raise_from
    raise e
HPOneViewException: Failure during login attempt.
vagrant@ecosystem-centos7:~/dev/python-hpOneView/examples  (enhancement/catch-http-lib-exceptions-as-hponeviewexceptions)

```
On Python3:
```
$ python3 fc_networks.py
Traceback (most recent call last):
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 485, in login
    self.validateVersion()
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 80, in validateVersion
    version = self.get(uri['version'])
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 325, in get
    resp, body = self.do_http('GET', uri, '')
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 127, in do_http
    conn.request(method, path, body, http_headers)
  File "/usr/local/lib/python3.5/http/client.py", line 1107, in request
    self._send_request(method, url, body, headers)
  File "/usr/local/lib/python3.5/http/client.py", line 1152, in _send_request
    self.endheaders(body)
  File "/usr/local/lib/python3.5/http/client.py", line 1103, in endheaders
    self._send_output(message_body)
  File "/usr/local/lib/python3.5/http/client.py", line 934, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.5/http/client.py", line 877, in send
    self.connect()
  File "/usr/local/lib/python3.5/http/client.py", line 1253, in connect
    super().connect()
  File "/usr/local/lib/python3.5/http/client.py", line 849, in connect
    (self.host,self.port), self.timeout, self.source_address)
  File "/usr/local/lib/python3.5/socket.py", line 712, in create_connection
    raise err
  File "/usr/local/lib/python3.5/socket.py", line 703, in create_connection
    sock.connect(sa)
TimeoutError: [Errno 110] Connection timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "fc_networks.py", line 54, in <module>
    oneview_client = OneViewClient(config)
  File "/home/vagrant/dev/python-hpOneView/hpOneView/oneview_client.py", line 119, in __init__
    self.__connection.login(config["credentials"])
  File "/home/vagrant/dev/python-hpOneView/hpOneView/connection.py", line 487, in login
    raise_from(HPOneViewException('Failure during login attempt.'), e)
  File "/usr/local/lib/python3.5/site-packages/future/utils/__init__.py", line 398, in raise_from
    exec(execstr, myglobals, mylocals)
  File "<string>", line 1, in <module>
hpOneView.exceptions.HPOneViewException: Failure during login attempt.
```

Python 2 ends up having a lot less understandable trace and message in this case

Reply:
So, after trying both the `reraise` method from `six` and the `raise_from` method from `future.utils`, both which are basically supposed to achieve [PEP 3134](https://www.python.org/dev/peps/pep-3134/) on Py2/Py3, I'm still seeing the same thing happening.

Py2 exception will remain not chained while Py3 has the chaining.

Do we want to keep this behavior @tmiotto ? Thoughts?

Another possibility: Specifically target `OSerror` as a `no route to host` and `TimeOut` Exception types, as those have been proven to be hit, and leave the rest to chance.

That way we do get better error messages, but we also do get more lines of code and still allow whatever other exception to pass through.

Both feel like tradeoffs to me, ideally the `from_raise` should've worked 😞